3rdParty/fuerte/src/message.cpp

```diff
@@ -32,7 +32,19 @@
 
 namespace {
 static std::string const emptyString;
+
+inline int hex2int(char ch) {
+  if ('0' <= ch && ch <= '9') {
+    return ch - '0';
+  } else if ('A' <= ch && ch <= 'F') {
+    return ch - 'A' + 10;
+  } else if ('a' <= ch && ch <= 'f') {
+    return ch - 'a' + 10;
+  }
+
+  return 0;
```

Comment:
The worst thing that can happen here is that we return `0` from `hex2int()`, which would be indistiguishable from a `%00`. What I can do is to throw an exception in case an invalid character is encountered, but that will mean invalid encodings may end up with an exception or an HTTP 500 error.